doc/source/user_guide/fields/field_data.rst

```diff
@@ -147,98 +161,34 @@ First, create a transaction object for field data.
+as fetching field data. You may either reuse the existing request object or create a new one
+by copying from a similar type:
 
-``payload_data`` is a dictionary containing the requested fields as a numpy array in the following order:
+.. code-block:: python
 
-``tag -> surface_id [int] -> field_name [str] -> field_data[np.array]``
+  >>> scalar_field_data = payload_data.get_field_data(scalar_request_1)
+  >>> scalar_field_data.keys()
+  dict_keys([1, 2])
+  >>> scalar_request_1 = scalar_request_1._replace(surfaces=[1])
+  >>> scalar_field_data = payload_data.get_field_data(scalar_request_1)
+  >>> scalar_field_data.keys()
+  dict_keys([1])
 
 .. note::
-   ``get_fields`` call also clears all requests, so that subsequent calls to this method
-   yield nothing until more requests are added.
-
-
-Tag
----
-A tag is tuple of input, value pairs for which field data is generated.
-
-For example, if you request the scalar field data for element location, in the
-dictionary, ``tag`` is ``(('type','scalar-field'), ('dataLocation', 1), ('boundaryValues',False))``.
-Similarly, if you request the boundary values for node location, ``tag`` is
-``(('type','scalar-field'), ('dataLocation', 0), ('boundaryValues',True)``.
-
-Surface ID
-----------
-The surface ID is the same one that is passed in the request.
-
-Field name
-----------
-A request returns multiple fields. The number of fields depends on the request
-type.
-
-Surface request
-~~~~~~~~~~~~~~~
-The response to a surface request contains any of the following fields,
-depending on the request arguments:
-
-- ``faces``, which contain face connectivity
-- ``vertices``, which contain node coordinates
-- ``centroid``, which contains face centroids
-- ``face-normal``, which contains face normals
-
-Scalar field request
-~~~~~~~~~~~~~~~~~~~~
-The response to a scalar field request contains a single field with the same
-name as the scalar field name passed in the request.
-
-Vector field request
-~~~~~~~~~~~~~~~~~~~~
-The response to a vector field request contains two fields:
-
-- ``vector field``, with the same name as the vector field name that is passed in the request.
-
-- ``vector-scale``, a float value indicating the vector scale.
-
-Pathlines field request
-~~~~~~~~~~~~~~~~~~~~~~~
-The response to a pathlines field request contains the following fields:
-
-- ``pathlines-count``, which contains pathlines count.
-- ``lines``, which contain pathlines connectivity.
-- ``vertices``, which contain node coordinates.
-- ``field name``, which contains pathlines field. field name is the same name as
-  the scalar field name passed in the request.
-- ``particle-time``, which contains particle time, if requested.
-- ``additional field name``, which contains additional field, if requested.
-  additional field name is the same name as the additional field name passed in
-  the request.
+  All request objects must be unique when adding requests or fetching data from a transaction.
```

Comment:
@prmukherj I think we should make a clearer, more concrete statement than this. What is the actual constraint?

Reply:
The constraint is one should not create duplicate requests and add them. or in that case the data will be wrongly fetched.